ledger/complete/mtrie/node/node.go

```diff
@@ -29,16 +29,28 @@ import (
 // for performance reasons, we not not copy read.
 // TODO: optimized data structures might be able to reduce memory consumption
 type Node struct {
+	// Implementation Comments:
+	// Formally, a tree can hold up to 2^maxDepth number of registers. However,
+	// the current implementation and is designed to operate on a sparsely populated
+	// tree, holding much less than 2^64 registers.
+
 	lChild    *Node           // Left Child
 	rChild    *Node           // Right Child
 	height    int             // height where the Node is at
 	path      ledger.Path     // the storage path (leaf nodes only)
 	payload   *ledger.Payload // the payload this node is storing (leaf nodes only)
 	hashValue []byte          // hash value of node (cached)
-	maxDepth  uint16          // captures the longest path from this node to compacted leafs in the subtree
-	regCount  uint64          // number of registers allocated in the subtree
+	// TODO : Atm, we don't support trees with dynamic depth.
+	//        Instead, this should be a forest-wide constant
+	maxDepth uint16 // captures the longest path from this node to compacted leafs in the subtree
+	// TODO : migrate to book-keeping only in the tree root.
+	//        Update can just return the _change_ of regCount.
+	regCount uint64 // number of registers allocated in the subtree
 }
 
+// why depth is in 16 bits?
+// is 64 bits enough for regCount?
+
```

Comment:
```suggestion
```
I added the comment above https://github.com/onflow/flow-go/blob/58e9ce49bfc76428e3be04941b1963385bb6b020/ledger/complete/mtrie/node/node.go#L32-L35 to answer this question here. Hence I suggested in my PR to remove this. 

To be more explicit: 
Even if every register only held a single byte, 2^64 registers would be more than an exabyte of data. As the mTrie is designed to be held in-memory, this is well beyond currently practical limitations (by at least 6 orders of magnitude).

Reply:
That makes sense!